include/phasar/PhasarLLVM/Pointer/LLVMPointsToSet.h

```diff
@@ -38,22 +39,27 @@ namespace psr {
 
 class LLVMPointsToSet : public LLVMPointsToInfo {
 private:
-  using PointsToSetMap = std::unordered_map<
-      const llvm::Value *,
-      std::shared_ptr<std::unordered_set<const llvm::Value *>>>;
+  using PointsToSetTy = std::unordered_set<const llvm::Value *>;
+  using PointsToSetPtrTy = std::shared_ptr<PointsToSetTy>;
```

Comment:
I agree, these `shared_ptr`s probably have non-negligible overhead that we should try to remove

Reply:
I have tried switching from `std::shared_ptr<PointsToSetTy>` to `PointsToSetTy *`, but I did not see much improvement - when badly implemented, it could even slow down the analysis.

instead, I have changed `std::shared_ptr<std::unordered_set<const llvm::Value *>>` to `std::shared_ptr<llvm::DenseSet<const llvm::Value *>>` which gave more speedup (30-40 seconds less on openssl)

Reply:
I still would prefer `unique_ptr` as they make reasoning about memory allocation much easier that `shared_ptr`, especially regarding lifetime.
Could you try a `unique_ptr` version? If you have one, I could take a look at it and maybe fix/improve things further.

Reply:
I have now pushed the initial `unique_ptr` version. The performance is slightly worse than before (only noticeable on larger targets such as OpenSSL, where it ran about 30s slower)